nava/functions.py

```diff
@@ -5,17 +5,46 @@
 import os
 import shlex
 from functools import wraps
-import threading
-
+from .thread import NavaThread
 from .params import OVERVIEW
 from .params import SOUND_FILE_PLAY_ERROR, SOUND_FILE_EXIST_ERROR
-from .params import SOUND_FILE_PATH_TYPE_ERROR
+from .params import SOUND_FILE_PATH_TYPE_ERROR, SOUND_ID_EXIST_ERROR
 from .errors import NavaBaseError
+from . import params
 
-"""
-List of all aplay processes
-"""
-play_processes = []
+
+def stop(sound_id):
+    """
+    Stop sound.
+
+    :param sound_id: sound id
+    :type sound_id: int
+    :return: None
+    """
+    if sound_id not in params._play_threads_map:
+        raise NavaBaseError(SOUND_ID_EXIST_ERROR)
+    params._play_threads_map[sound_id].stop()
+
+
+def stop_all():
+    """
+    Stop all sounds.
+
+    :return: None
+    """
+    for thread in params._play_threads_map.values():
+        thread.stop()
+
+
+def sound_id_gen():
+    """
+    Sound id generator.
+
+    :return: sound id as int
+    """
+    params._play_threads_counter += 1
+    sound_id = params._play_threads_counter + 1000
```

Comment:
Why did you add `1000` offset. If it's necessary, what do you think of transferring it to `params`.

Reply:
For better readability, we may change the format of the `sound_id` to something like `Nava-1`, `Nava-2`, etc. in the future.